src/pages/api/v1/bookmarks/add/data.tsx

```diff
@@ -0,0 +1,211 @@
+
+type MinDataApiResponse = {
+	data: ApiResponse;
+	status: number;
+};
+
+/**
+ * Helper function to call the min-data API
+ */
+const callMinDataApi = async (
+	bodyData: z.infer<ReturnType<typeof getBookmarkBodySchema>>,
+	cookies: { [key: string]: string },
+): Promise<MinDataApiResponse> => {
+	try {
+		const minDataResponse = await axios.post(
+			`${getBaseUrl()}${NEXT_API_URL}${ADD_BOOKMARK_MIN_DATA_API}`,
+			bodyData,
+			{
+				headers: {
+					Cookie: apiCookieParser(cookies),
+				},
+			},
+		);
+
+		return {
+			status: minDataResponse.status,
+			data: minDataResponse.data,
+		};
+	} catch (error) {
+		const errorMessage = formatErrorMessage(error);
+		Sentry.captureException("Error calling min-data API", {
+			extra: { error: errorMessage },
+		});
+		return {
+			status: 500,
+			data: {
+				error: "Failed to process bookmark",
+				message: errorMessage,
+				data: null,
+			},
+		};
+	}
+};
+
+/**
+ * /api/v1/bookmarks/add/data:
+ *   post:
+ *     summary: Add a new bookmark
+ *     description: Main endpoint to add a new bookmark. Orchestrates min-data, screenshot and remaining data processing.
+ *     tags:
+ *       - Bookmarks
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             required:
+ *               - url
+ *               - category_id
+ *               - update_access
+ *             properties:
+ *               url:
+ *                 type: string
+ *                 description: URL to bookmark
+ *               category_id:
+ *                 type: number
+ *                 description: Category ID to add bookmark to
+ *               update_access:
+ *                 type: boolean
+ *                 description: Whether user has update access
+ *     responses:
+ *       200:
+ *         description: Bookmark added successfully
+ *       400:
+ *         description: Invalid request body
+ *       401:
+ *         description: Unauthorized
+ *       403:
+ *         description: Forbidden - User does not have update access
+ *       409:
+ *         description: Bookmark already exists in category
+ *       500:
+ *         description: Internal server error
+ */
+export default async function handler(
+	request: NextApiRequest<AddBookmarkMinDataPayloadTypes>,
+	response: NextApiResponse<ApiResponse>,
+) {
+	const supabase = apiSupabaseClient(request, response);
+
+	if (request.method !== "POST") {
+		response.status(405).send({
+			error: "Only POST requests allowed",
+			message: "Only POST requests allowed",
+			data: null,
+		});
+		return;
+	}
+
+	// Validate request body
+	const schema = getBookmarkBodySchema();
+	let bodyData;
+	try {
+		bodyData = schema.parse(request.body);
+	} catch (parseError) {
+		const errorMessage = formatErrorMessage(parseError);
+
+		response.status(400).send({
+			error: "Invalid request body",
+			message: errorMessage,
+			data: null,
+		});
+		return;
+	}
```

Comment:
To avoid the let and try catch, can we use the `safeParse`

<img width="819" height="471" alt="Image" src="https://github.com/user-attachments/assets/4b5b9222-6543-4160-ab49-8fa774ab9bfc" />